js/plugins/iq.js

```diff
@@ -0,0 +1,348 @@
+                     " but ooyalaReporter is: " + this.ooyalaReporter);
+            }
+          }
+        }
+        break;
+
+      case OO.Analytics.EVENTS.VIDEO_PAUSED:
+        this.ooyalaReporter.reportPause();
+        OO.log("IQ: Reported: reportPause()");
+        break;
+
+      // TODO: use for resume?
+      case OO.Analytics.EVENTS.VIDEO_PLAYING:
+        this.ooyalaReporter.reportResume();
+        OO.log("IQ: Reported: reportResume()");
+        break;
+
+      case OO.Analytics.EVENTS.VIDEO_SEEK_COMPLETED:
+        if (params && params[0])
+        {
+          var seekedPlayheadPosition = params[0].timeSeekedTo;
+          var seekedPlayheadPositionMilli = seekedPlayheadPosition * 1000;
+          var currentPlayheadPositionMilli = currentPlayheadPosition * 1000;
+          this.ooyalaReporter.reportSeek(currentPlayheadPositionMilli, seekedPlayheadPositionMilli);
+          OO.log("IQ: Reported: reportSeek() with args: " + currentPlayheadPositionMilli + ", " + seekedPlayheadPositionMilli);
+        }
+        break;
+
+      case OO.Analytics.EVENTS.PLAYBACK_COMPLETED:
+        this.ooyalaReporter.reportComplete();
+        OO.log("IQ: Reported: reportComplete()");
+        break;
+
+      case OO.Analytics.EVENTS.VIDEO_REPLAY_REQUESTED:
+        this.ooyalaReporter.reportReplay();
+        OO.log("IQ: Reported: reportReplay()");
+        break;
+
+      case OO.Analytics.EVENTS.AD_REQUEST:
+      case OO.Analytics.EVENTS.AD_REQUEST_SUCCESS:
+      case OO.Analytics.EVENTS.AD_SDK_LOADED:
+      case OO.Analytics.EVENTS.AD_SDK_LOAD_FAILURE:
+      case OO.Analytics.EVENTS.AD_BREAK_STARTED:
+      case OO.Analytics.EVENTS.AD_BREAK_ENDED:
+      case OO.Analytics.EVENTS.AD_POD_STARTED:
+      case OO.Analytics.EVENTS.AD_POD_ENDED:
+      case OO.Analytics.EVENTS.AD_STARTED:
+      case OO.Analytics.EVENTS.AD_ENDED:
+      case OO.Analytics.EVENTS.AD_SKIPPED:
+      case OO.Analytics.EVENTS.AD_ERROR:
+      case OO.Analytics.EVENTS.AD_REQUEST_EMPTY:
+      case OO.Analytics.EVENTS.AD_REQUEST_ERROR:
+      case OO.Analytics.EVENTS.AD_PLAYBACK_ERROR:
+      case OO.Analytics.EVENTS.AD_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_SDK_IMPRESSION:
+      case OO.Analytics.EVENTS.AD_COMPLETED:
+      case OO.Analytics.EVENTS.AD_CLICKTHROUGH_OPENED:
+      case OO.Analytics.EVENTS.SDK_AD_EVENT:
+        var eventMetadata = params[0];
+        if(!eventMetadata)
+        {
+          eventMetadata = {};
+        }
+
+        if (eventMetadata.adEventName)
+        {
+          eventMetadata.adEventName = eventName + ":" + eventMetadata.adEventName;
+        }
+        else
+        {
+          eventMetadata.adEventName = eventName;
+        }
+        this.ooyalaReporter.reportCustomEvent(eventName, eventMetadata);
+        OO.log("IQ: Reported: reportCustomEvent() for event: " + eventName + " with args:" + JSON.stringify(eventMetadata));
```

Comment:
You don't have to fix this but our console is filled with a bunch of uncaught exceptions from JSON.stringify and self referencing variables. Do you know of a fix?

Reply:
I don't know specifically but I did see some issues with some objects (particularly when the ad object from ima was being sent). Would a simple try / catch work? I can try to experiment a bit later on